modal/shared/volumes.py

```diff
@@ -10,7 +10,9 @@ def get_model_path(model_name: str):
     return models_path / model_name.lower()
 
 
-def does_model_exist(model_path: Path):
+# gets passed a str in some circumstances despite being typed as Path
+def does_model_exist(model_path: Path | str):
+    model_path = Path(model_path)
```

Comment:
Might need to revert this (and the model_path passing down to our container) to just string. When we instantiate a GPU container, the props passed down will be serialized by the Modal abstraction, using pickle, and afaik it will not be deserialized on our code properly. 

3 options:

1. Ping Modal about better type-casting, to which I think their best attempt would be a pydantic pipeline, since from their POV this param has arbitrary type. 
2. Make a serialize/deserialize pipeline, which might not work with modal parameterized req
3. Stick to passing down primitive types (revert to string)